### ISSUE TITLE
forge addon - add endpoint should not reformat code

### ISSUE THREAD

**Issue by davsclaus** [fabric8io/fabric8#5444]
like the edit endpoint no longer does when its editing in the route builder configure method.

**Comment by davsclaus** [comment]
The edit no longer reformat the code because we do a search/replace on the line. However for adding new code, then its roaster/forge doing the re-format.

@gastaldi wonder if forge has any hooks into Eclipse / IDEA etc to let it run its re-format "logic" so when we add new code to java code using roaster, then we can run the IDEA re-format afterwards so the code is "what the end user" would do if he run the reformat command manually in IDEA.

**Comment by gastaldi** [comment]
Roaster has a `toUnformattedString()` that does not attempt to reformat the parsed contents. Would that help?

**Comment by davsclaus** [comment]
Yes we use it in some use-cases. But when you add new annotations / fields / imports etc, then there is a alot more code to add. 

The unformatted works fine when I know its just a string in a method that is being changed, then i can use the unformatted to find the old string and replace it.

See here where we track if there is _new code being added_ or not
https://github.com/fabric8io/fabric8-forge/blob/master/addons/camel/src/main/java/io/fabric8/forge/camel/commands/project/ConfigureEndpointPropertiesStep.java#L476

And then change the source code accordingly.

**Comment by davsclaus** [comment]
@gastaldi  I wonder if roaster can be improved so if you add a field, then it only adds the field and do not re-write / re-format then entire java source code.

I guess its a hard problem to track, but you end up loosing comments and reformatting all your code and whatnot, which is not a pleasant experience.

For example adding a field with an annotation and optionally add missing java imports should be possible without touching any of the other source code.

**Comment by gastaldi** [comment]
Roaster uses JDT which in turn relies on AST. Comments should not be lost
in the AST node.

I expect the toUnformattedString() method would do what you are looking
for, otherwise we'd have to review it.
Em 29/02/2016 06:42, "Claus Ibsen" <notifications@github.com> escreveu:
